ecr/ref.go

```diff
@@ -70,6 +74,60 @@ func (spec ECRSpec) Registry() string {
 	return spec.arn.AccountID
 }
 
+// ParseEcrImageNameToRef takes an ECR image name and returns its canonical `ref` format.
+// i.e. ecr.aws/arn:<partition>:ecr:<region>:<account>:repository/<name>:<tag>
+func ParseEcrImageNameToRef(input string) (string, error) {
```

Comment:
minor: I think I'd prefer if this became a constructor to `ECRSpec`.  The `ECRSpec` object contains all the information and can be rendered into a ref by its `Canonical` method.